python/ray/_private/runtime_env/packaging.py

```diff
@@ -784,7 +785,20 @@ async def download_and_unpack_package(
                             "`pip install boto3` to fetch URIs in s3 "
                             "bucket. " + install_warning
                         )
-                    tp = {"client": boto3.client("s3")}
+                    s3_kwargs = {}
+                    if runtime_env:
+                        env_vars = runtime_env.env_vars()
+                        if "AWS_ENDPOINT_URL" in env_vars:
+                            s3_kwargs["endpoint_url"] = env_vars["AWS_ENDPOINT_URL"]
+                        if "AWS_ACCESS_KEY_ID" in env_vars:
+                            s3_kwargs["aws_access_key_id"] = env_vars[
+                                "AWS_ACCESS_KEY_ID"
+                            ]
+                        if "AWS_SECRET_ACCESS_KEY" in env_vars:
+                            s3_kwargs["aws_secret_access_key"] = env_vars[
+                                "AWS_SECRET_ACCESS_KEY"
+                            ]
+                    tp = {"client": boto3.client("s3", **s3_kwargs)}
```

Comment:
Per https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html, we can save credentials to `~/.aws/credentials` and everything should work?